### PR TITLE
[MI-2281] Add test cases for client.go and http.go

### PR DESCRIPTION
**Pull request by ayusht2810**

### CODE REVIEW

File: server/plugin/client_test.go [hunk 1]
```diff
@@ -432,3 +432,182 @@ func TestGetRecordFromServiceNowClient(t *testing.T) {
...
+	for _, testCase := range []struct {
+		description string
+		statusCode  int
+		setupClient func(c *client)
+		expectedErr string
+	}{
+		{
+			description: "GetAllComments: valid",
+			setupClient: func(c *client) {
+				monkey.PatchInstanceMethod(reflect.TypeOf(c), "CallJSON", func(_ *client, _, _ string, _, _ interface{}, _ url.Values) (_ []byte, _ int, _ error) {
```
Comment (Kshitij-Katiyar) [inline server/plugin/client_test.go:448]: if you patching same function for all test cases its better to do this once and you can have statusCode as a variable and can return this from monkey patching

**Review by Kshitij-Katiyar — APPROVED** [review]

File: server/plugin/http_test.go [hunk 2]
```diff
@@ -41,3 +44,87 @@ func TestCallJSON(t *testing.T) {
...
+			setupClient: func(c *client) {
+				monkey.PatchInstanceMethod(reflect.TypeOf(c.httpClient), "Do", func(*http.Client, *http.Request) (*http.Response, error) {
+					return &http.Response{}, errors.New("mockError invalid character '<'")
+				})
+			},
+			expectedErrorMessage: ErrorContentTypeNotJSON.Error(),
+			expectedStatusCode:   http.StatusInternalServerError,
+		},
+		{
+			description: "Call: Do method returns an error 2",
```
Comment (manojmalik20) [inline server/plugin/http_test.go:73]: Change this description.

**Review by manojmalik20 — APPROVED** [review]

File: server/plugin/client.go [hunk 3]
```diff
@@ -23,7 +23,7 @@ type Client interface {
 	CheckForDuplicateSubscription(*serializer.SubscriptionPayload) (bool, int, error)
 	SearchRecordsInServiceNow(tableName, searchTerm, limit, offset string) ([]*serializer.ServiceNowPartialRecord, int, error)
 	GetRecordFromServiceNow(tableName, sysID string) (*serializer.ServiceNowRecord, int, error)
-	GetAllComments(recordType, recordID string) (string, int, error)
+	GetAllComments(recordType, recordID string) (*serializer.ServiceNowComment, int, error)
```
[the remark this answers is not in the document]
Comment (ayusht2810) [reply server/plugin/client.go:26]: we were returning `comments.Result.CommentsAndWorkNotes` from the method which was throwing an error of nil pointer dereference.

[the remark this answers is not in the document]
Comment (manojmalik20) [reply server/plugin/client.go:26]: Okay.

**Pull request merged**